Tribler/Core/Modules/MetadataStore/OrmBindings/channel_metadata.py

```diff
@@ -0,0 +1,259 @@
+            Use it to consolidate fragmented channel torrent directories.
+            :param key: The public/private key, used to sign the data
+            """
+
+            self.commit_channel_torrent()
+
+            folder = os.path.join(self._channels_dir, self.dir_name)
+            for filename in os.listdir(folder):
+                file_path = os.path.join(folder, filename)
+                if filename.endswith(BLOB_EXTENSION):
+                    os.unlink(file_path)
+                else:
+                    raise ValueError
+            self.update_channel_torrent(self.contents_list)
+
+        def update_channel_torrent(self, metadata_list):
+            """
+            Channel torrents are append-only to support seeding the old versions
+            from the same dir and avoid updating already downloaded blobs.
+            :param metadata_list: The list of metadata entries to add to the torrent dir
+            :return The new infohash, should be used to update the downloads
+            """
+            # Create dir for metadata files
+            channel_dir = os.path.abspath(os.path.join(self._channels_dir, self.dir_name))
+            if not os.path.isdir(channel_dir):
+                os.makedirs(channel_dir)
+
+            new_version = self.version
+
+            # Write serialized and signed metadata into files
+            for metadata in metadata_list:
+                new_version += 1
+                with open(os.path.join(channel_dir, str(new_version).zfill(9) + BLOB_EXTENSION), 'wb') as f:
+                    serialized = metadata.serialized_delete() if metadata.deleted else metadata.serialized()
+                    f.write(''.join(serialized))
+
+            # Make torrent out of dir with metadata files
+            torrent, infohash = create_torrent_from_dir(channel_dir, os.path.join(self._channels_dir, self.dir_name + ".torrent"))
+
+            self.update_metadata(update_dict={"infohash": infohash, "version": new_version,
+                                              "torrent_date": datetime.utcfromtimestamp(torrent['creation date'])})
+
+            # Write the channel mdblob away
+            with open(os.path.join(self._channels_dir, self.dir_name + BLOB_EXTENSION), 'wb') as out_file:
+                out_file.write(''.join(self.serialized()))
+            return infohash
+
+        def commit_channel_torrent(self):
+            """
+            Collect new/uncommitted and marked for deletion metadata entries, commit them to channel torrent and
+            remove the obsolete entries if the commit succeeds.
+            :return The new infohash, should be used to update the downloads
+            """
+            new_infohash = None
+            try:
+                new_infohash = self.update_channel_torrent(self.staged_entries_list)
+            except IOError:
+                print ("Error during channel torrent commit, not going to garbage collect the channel")
```

Comment:
Please avoid using print statements. Instead, you should use the `logging` framework.